tests/impls/gen_object_detector_blackbox_sal/test_occlusion_based.py

```diff
@@ -143,13 +127,18 @@ def detect_objects(
             'xaitk_saliency.impls.gen_object_detector_blackbox_sal.occlusion_based.occlude_image_batch',
             wraps=occlude_image_batch
         ) as m_occ_img:
-            inst = PerturbationOcclusion(test_pi, test_gen)
-            inst.fill = test_fill
-            test_result = inst._generate(
+            m_impl = mock.Mock(spec=PerturbationOcclusion)
+            m_impl._perturber = _perturb
+            m_impl._generator = _generate
+            m_impl.fill = test_fill
+            m_impl._threads = 0
+
+            test_result = PerturbationOcclusion._generate(
```

Comment:
This seems weird to me. We're mocking _generate, but also testing it? Why aren't we calling this on `m_impl` directly? I don't think mocking should prevent that